Reflects the AB button function rearranged on the left and right.

okawari10 opened the pull request:
Reflects the AB button function rearranged on the left and right.
Prohibition of double tap ZOOM.
Forward and Backward function during pause.
Aspect ratio re-adjustment.
Re-considering the red frame and green frame that specifies the video control.

okawari10 committed:
Reflects the AB button function rearranged on the left and right.
Prohibition of double tap ZOOM.
Forward and Backward function during pause.
Aspect ratio re-adjustment.
Re-considering the red frame and green frame that specifies the video control.

mkx76 commented:
確認環境
- MBP/Chrome
- Pixel4/Chrome

- 右側で動画を選択しても表示／再生できない
- 青色バー押下で選択していない際でも反対側の「A-B」が押せてしまうので動作しないならdisableにしたほうがいい
- 青色バー押下で選択になるが、枠がわかりにくい。今回操作パネルも枠で囲われているので、背景色を変えるなどのほうがわかりやすいかも
- 画面の選択状態によって`|◀`  `▶|`を押しても反応しないときがある。機能が動作しないならdiableしたほうがいい
-  `▶▶` 実施中に `◀◀` を押下すると以下のようなエラーが発生する
```shellscript
Uncaught DOMException: Failed to set the 'playbackRate' property on 'HTMLMediaElement': The provided playback rate (-1) is not in the supported playback range.
```
- `◀◀`を押下しても逆再生になっていない（順再生になっている）
- 再生時になにかのタイミングで再生箇所が終点に固定されて、そこからどうやってもそのタイミングの再生箇所から変更できない
- 動作再生中の`▶|`押下時の正しい動きは？
- `▶▶`を何度も押すとrange外になってエラーになる
```shellscript
Uncaught DOMException: Failed to set the 'playbackRate' property on 'HTMLMediaElement': The provided playback rate (17) is not in the supported playback range.
```

全体的に動作が不確定というか不安定です。
ブランチのコードだけで期待どおりの動作をするのか一度確認をお願いします。

mkx76 commented:
今回縦画面は確認対象外でよかったですか？
- 縦画面で青色バーをタップしても選択枠が表示されない、AB区間選択などができない

okawari10 committed:
Add gitignore.

okawari10 committed:
bug-fix of following three items:
右側で動画を選択しても表示／再生できない.
青色バー押下で選択していない際でも反対側の「A-B」が押せてしまうので動作しないならdisableにしたほうがいい.
青色バー押下で選択になるが、枠がわかりにくい。今回操作パネルも枠で囲われているので、背景色を変えるなどのほうがわかりやすいかも.

okawari10 commented:
指摘の一部（下記）を先ずは修正しましたので確認お願いいたします。
- 右側で動画を選択しても表示／再生できない
- 青色バー押下で選択になるが、枠がわかりにくい。今回操作パネルも枠で囲われているので、背景色を変えるなどのほうがわかりやすいかも

okawari10 committed:
Fixed for the following point out:
青色バー押下で選択していない際でも反対側の「A-B」が押せてしまうので動作しないならdisableにしたほうがいい

okawari10 commented:
下記も対応しました。
青色バー押下で選択していない際でも反対側の「A-B」が押せてしまうので動作しないならdisableにしたほうがいい
b_rangeAB[ ].disabled = true / false; を追加

okawari10 committed:
下記指摘に対してplay中に|◀ ▶|押された場合はpauseしてから処理を実行するように変更。
画面の選択状態によって|◀ ▶|を押しても反応しないときがある。機能が動作しないならdiableしたほうがいい

okawari10 commented:
PCで ◀◀ を押下するとエラーが発生するようで、playbackRateが負の値に（逆再生）できるのはスマホのみのようです。
https://gray-code.com/javascript/reverse-play-the-video/
|◀と同様にcurrentTimeを毎秒戻すように変更した方が良いか？

okawari10 committed:
ループ再生追加、早送り、早戻し機能をコマ送りと同様の挙動にした。ただし送る時間を進めている。（PCでplaybackRateを（巻き戻し）負の値にした時にエラーが出るため）
